Automattic/woocommerce-payments
# Fix the link to customers table

Fixes #3037 

#### Changes proposed in this Pull Request

**WooCommerce -> customers** table reads records from `wp_wc_customers_lookup` table but when building link from `includes/wc-payment-api/class-wc-payments-api-client.php` we are passing id that is read from `wp_users` table. 

We're using `$order->get_customer_id()` method but it returns id from `wp_users` table. If guest checkouts are turned off this causes no visual harm as id's in both tables are coincide but once guess checkout is allowed two potential problems may occur:
- For guess customers given method returns `0`
- For non-guess customers method can return incorrect id if there have been previous guest checkouts

The proposed solution uses an available method from the WC-Admin plugin that takes care of fetching the correct customer id.

During the investigation, we found another issue. With Analytics disabled we don't create entries in `wp_wc_customers_lookup` table but once Analytics is enabled we are trying to link to the non-existing customer. The solution to this problem is simply not displaying the entry as link if we can't find customer id.

#### Testing instructions

1. Make sure that Accounts & Privacy settings for WooCommerce allow for guest checkout
2. Make a purchase as a guest
3. Make a purchase from an account (or create during checkout)
4. Make sure that `wc-admin_import_customers` scheduled action from **WooCommerce -> status** are run
4. Go to **Payments -> transactions*
5. Check that for both transactions the customer link points to the correct customer
6. Go to **WooCommerce -> settings** 'Advanced' tab under 'features' uncheck "Enable Analytics".
7. Repeat steps 2. - 5. 

-------------------
